app/renderer/components/preferences/pluginsTab.js

```diff
@@ -24,25 +26,35 @@ const {SettingsList, SettingCheckbox} = require('../common/settings')
 const {DefaultSectionTitle} = require('../common/sectionTitle')
 
 class PluginsTab extends ImmutableComponent {
+  constructor (props) {
+    super(props)
+    this.onToggleFlash = this.onTogglePlugin.bind(this, 'flash')
+    this.onToggleWidevine = this.onTogglePlugin.bind(this, 'widevine')
+  }
+
   get flashInstalled () {
     return getSetting(settings.FLASH_INSTALLED, this.props.settings)
   }
 
-  onToggleFlash (e) {
-    aboutActions.setResourceEnabled(flash, e.target.value)
-    if (e.target.value !== true) {
-      // When flash is disabled, clear flash approvals
-      aboutActions.clearSiteSettings('flash', {
-        temporary: true
-      })
-      aboutActions.clearSiteSettings('flash', {
-        temporary: false
-      })
+  onTogglePlugin (plugin, e) {
+    switch (plugin) {
+      case 'widevine':
+        aboutActions.setResourceEnabled(widevine, e.target.value)
+        break
+      case 'flash':
+        aboutActions.setResourceEnabled(flash, e.target.value)
+        if (e.target.value !== true) {
+          // When flash is disabled, clear flash approvals
+          aboutActions.clearSiteSettings('flash', {
+            temporary: true
+          })
+          aboutActions.clearSiteSettings('flash', {
+            temporary: false
+          })
+        }
+        break
     }
-  }
-
-  onToggleWidevine (e) {
-    aboutActions.setResourceEnabled(widevine, e.target.value)
+    ipc.send(messages.PREFS_RESTART)
```

Comment:
i think it's slightly better to move this and the other `ipc.send(messages.PREFS_RESTART)` into an action in `js/about/aboutActions`

Reply:
@diracdeltas sounds like a good idea. I can include that in to an action in this PR. Should I move all of the other restart notices in the codebase to the new aboutAction as well, or do that in a separate PR? Thanks!

Reply:
@ryanml you can do it all in this PR, thanks